text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,799 @@
+
+- The Resource Coordinator will detect conflicts in Entity of the same type being discovered and choose one to use.
+- When using Entity Detectors and Resource detectors together, the following merge rules will be used:
+  - Entity merging will occur first resulting in an "Entity Merged" Resource (See [algorithm here](#entity-merging-and-resource)).
+  - Resource detectors otherwise follow existing merge semantics.
+    - The Specification merge rules will be updated to account for violations prevalent in ALL implementation of resource detection.
+    - Specifically: This means the [rules around merging Resource across schema-url will be dropped](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#merge).  Instead only conflicting attributes will be dropped.
+    - SchemaURL on Resource will need to be deprecated with entity-specific schema-url replacing it. SDKs will no longer fill out SchemaURL on Resource. Additionally, as no (non-service) Resource semantic conventions have ever stabilized, SchemaURL usage on Resource cannot be in stable components of OpenTelemetry.  Given prevalent concerns of implementations around Resource merge specification, we suspect impact of this deprecation to be minimal.
+  - An OOTB ["Env Variable Entity Detector"](#environment-variable-detector) will be specified and provided vs. requiring SDK wide ENV variables for resource detection.
+- *Additionally, Resource Coordinator would be responsible for understanding Entity lifecycle events, for Entities whose lifetimes do not match or exceed the SDK's own lifetime (e.g. browser session).*
+
+#### Entity Detector
+
+The Entity detector in the SDK is responsible for detecting possible entities that could identify the SDK.  For Example, if the SDK is running in a kubernetes pod, it may provide an Entity for that pod.   SDK Entity Detectors are only required to provide identifying attributes, but may provide descriptive attributes to ensure combined Resource contains similar attributes as today's SDK.
+
+An Entity Detector would have an API similar to:
+
+```rust
+trait EntityDetector
+  pub fn detect_entities(...) -> Result<Vec<Entity>, EntityDetectionError>>
+```
+
+Where `Result` is the equivalent of error channel in the language of choice (e.g. in Go this would be `entities, err := e.detectEntities()`).
+
+#### Entity Merging and Resource
+
+The most important aspect of this design is how Entities will be merged to construct a Resource.
+
+We provide a simple algorithm for this behavior:
+
+- Construct a set of detected entities, `E`
+  - All entity detectors are sorted by priority (highest first)
+  - For each entity detector `D`, detect entities
+    - For each entity detected, `d'`
+      - If an entity `e'` exists in `E` with same entity type as `d'`, do one of the following:
+        - If the entity identiy and schema_url are the same, merge the descriptive attributes of `d'` into `e'`:
```

Comment:
are omitted `schema_url`s identical? or do you force consumers into using schema_urls?

Reply:
omitted schema_urls are considered different.   Basically imagine a non-existent schema-url as some random ID that is disjoint from all possible other schema urls.

Reply:
I can accept both choices and have seen both in the past. Can we add a note on that to the section maybe citing semantic of schema_urls?

Reply:
If you haven't, I'd read the original resource merge specification linked to above - where that semantic/precedence is set.

Reply:
If you refer to (https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#merge), I've read it and it is, TBH, a required read to understand what "merging" is in this context. Nonetheless, the linked document isn't very specific. If you consider that the semantics of a schema_url, then I'd kindly ask you to add what you explained to the comment.

Reply:
this is a proposal against the otel specification.  The specification is required reading for understanding the context of the proposal.

If the linked document wasn't specific - it's an issue in the specification we need to improve.

Part of the art here though is NOT being specific.  Being too prescriptive can lead to an unimplementable specification, or one too tied to specific languages and unable to be used broadly.